examples/qfiae/utils_qfiae.py

```diff
@@ -0,0 +1,108 @@
+    for i in range(1, int((len(coeffs) - 1) / 2) + 1):
+        n += 1
+        series += coeffs[i] * np.exp(1j * n * x_vals * (2 * np.pi / period))
+    for i in range(int((len(coeffs) - 1) / 2 + 1), len(coeffs), 1):
+        series += coeffs[i] * np.exp(1j * -n * x_vals * (2 * np.pi / period))
+        n -= 1
+    y_vals = np.real(series)
+    return y_vals
+
+
+def plot_data_fourier(function_test, xtest, y_fourier):
+    """Plots the Fourier representation alongside the function test and provides the efficiency of the fit."""
+    SSE = 0
+    SST = 0
+    average = np.sum(function_test) / function_test.size
+    N = len(function_test)
+    f_pred = np.zeros(N, dtype="complex")
+
+    for i in range(N):
+        f_pred_x = y_fourier[i]
+        f_pred[i] = f_pred_x
+        SSE += (function_test[i] - f_pred_x) ** 2
+        SST += (function_test[i] - average) ** 2
+
+    R = 1 - SSE / SST
+    R = round(float(R) * 100, 1)
+
+    sort_indices = np.argsort(xtest)
+    plt.plot(xtest[sort_indices], function_test[sort_indices], label="Target")
+    plt.plot(
+        xtest[sort_indices],
+        y_fourier[sort_indices],
+        label=f"Quantum Fourier \nAccuracy: {R}%",
+    )
+
+    plt.xlabel("x")
+    plt.ylabel("f(x)")
+    plt.legend(loc="upper center")
+
+    return plt, R
+
+
+def exp_fourier_to_trig(fourier_coeffs):
+    """Converts the Fourier coefficients from exponential form to trigonometric form."""
+
+    n = len(fourier_coeffs)
+    c0 = fourier_coeffs[0]
+    b_coeffs = [2 * np.real(fourier_coeffs[i]) for i in range(1, int(n / 2) + 1)]
+    a_coeffs = [-2 * np.imag(fourier_coeffs[i]) for i in range(1, int(n / 2) + 1)]
+    return [c0] + b_coeffs + a_coeffs
+
+
+def coeffs_array_to_class(fourier_coeffs):
+    """Converts the Fourier coefficients from array to a Class to be dealt in an easier way."""
+
+    fourier_class = []
+
+    for i in range(len(fourier_coeffs)):
+        fourier_class_term = Fourier_terms(
+            fourier_coeffs[i, 2], int(fourier_coeffs[i, 1]), int(fourier_coeffs[i, 0])
+        )
+        fourier_class.append(fourier_class_term)
+    return fourier_class
+
+
+def trig_to_final_array(trig_coeffs):
+    """Converts the array with the trigonometric coefficients into an array that tracks if a
+    coefficient corresponds to a Cosine (0) or to a Sine (1)."""
+
+    array_list = np.zeros(shape=(len(trig_coeffs), 3))
+    array_list[0, 0] = 0  # 0 means Cos
+    array_list[0, 1] = 0
+    array_list[0, 2] = np.real(trig_coeffs[0])
+    for i in range(1, int((len(trig_coeffs) - 1) / 2) + 1):
+        # cosine
+        array_list[i, 0] = 0  # 0 means Cos
+        array_list[i, 1] = i
+        array_list[i, 2] = trig_coeffs[i]
+    for i in range(int((len(trig_coeffs) - 1) / 2) + 1, len(trig_coeffs)):
+        # sine
+        array_list[i, 0] = 1  # 1 means Sin
+        array_list[i, 1] = i - (len(trig_coeffs) - 1) / 2
+        array_list[i, 2] = trig_coeffs[i]
+    return array_list
+
+
+class Fourier_terms:

```

Comment:
I don't understand why you need a class here. 
Can you explain how do you use this?

Reply:
Thanks for the suggestions! I've already implemented some. 
Here I don't necessarily need this Class, of course, I can just work with the array, but for me, the code seems easier to understand with the notation introduced in `Fourier_terms`.  Since the coefficient of the Fourier term will be in the `self.coefficient` attribute, the angle in  `self.angle` and the type of function in `self.function`
On the other hand, I also see your point and it is probably better to remove unnecessary functions/classes. So, let me know if you want me to remove this.

Reply:
If you think it is clearer, feel free to keep it :)
Maybe you can think if it is possible to add some method to this class, maybe collecting some of the functions you write above. If it doesn't make sense, keep it as it its.